core/chains/evm/gas/models.go

```diff
@@ -103,7 +137,78 @@ type Estimator interface {
 	// attempts must:
 	//   - be sorted in order from highest price to lowest price
 	//   - all be of transaction type 0x2
-	BumpDynamicFee(ctx context.Context, original DynamicFee, gasLimit uint32, maxGasPriceWei *assets.Wei, attempts []PriorAttempt) (bumped DynamicFee, chainSpecificGasLimit uint32, err error)
+	BumpDynamicFee(ctx context.Context, original DynamicFee, gasLimit uint32, maxGasPriceWei *assets.Wei, attempts []EvmPriorAttempt) (bumped DynamicFee, chainSpecificGasLimit uint32, err error)
+}
+
+type EvmFee struct {
+	Legacy  *assets.Wei
+	Dynamic *DynamicFee
+}
+
+// FeeEstimator provides a generic interface for fee estimation
+//
+//go:generate mockery --quiet --name FeeEstimator --output ./mocks/ --case=underscore
+type FeeEstimator[HEAD any, FEE any, MAXPRICE any, HASH any] interface {
+	OnNewLongestChain(context.Context, HEAD)
+	Start(context.Context) error
+	Close() error
+
+	GetFee(ctx context.Context, calldata []byte, feeLimit uint32, maxFeePrice MAXPRICE, opts ...Opt) (fee FEE, chainSpecificFeeLimit uint32, err error)
+	BumpFee(ctx context.Context, originalFee FEE, feeLimit uint32, maxFeePrice MAXPRICE, attempts []PriorAttempt[FEE, HASH]) (bumpedFee FEE, chainSpecificFeeLimit uint32, err error)
+}
+
+// WrappedEvmEstimator provides a struct that wraps the EVM specific dynamic and legacy estimators into one estimator that conforms to the generic FeeEstimator
+type WrappedEvmEstimator struct {
+	EvmEstimator
+	EIP1559Enabled bool
+}
+
+// var _ FeeEstimator = (*WrappedEvmEstimator)(nil)
+var _ FeeEstimator[*evmtypes.Head, EvmFee, *assets.Wei, common.Hash] = (*WrappedEvmEstimator)(nil)
```

Comment:
what's the purpose of this? I've seen this pattern in a couple of files

Reply:
Is this just to verify that generic types are declared appropriately

Reply:
it's to make sure that `WrappedEvmEstimator` conforms to the interface defined in `FeeEstimator`